dashboard/test/controllers/home_controller_test.rb

```diff
@@ -194,4 +192,38 @@ def setup_user_with_gallery
       get :debug
     end
   end
+
+  test 'workshop organizers see only new dashboard links' do
+    sign_in create(:workshop_organizer, terms_of_service_version: 1)
+    get :home
+    assert_select 'h1', text: 'Workshop Dashboard'
+  end
+
+  test 'workshop admins see new and old dashboard links' do
+    sign_in create(:workshop_admin, terms_of_service_version: 1)
+    get :home
+    assert_select 'h1', text: 'Workshop Dashboard'
+    assert_select 'h1', text: 'Old CSF Workshop Dashboard'
+  end
+
+  test 'facilitators see only new dashboard links' do
+    facilitator = create(:facilitator, terms_of_service_version: 1)
+    sign_in facilitator
+    get :home
+    assert_select 'h1', text: 'Workshop Dashboard'
+  end
+
+  test 'legacy workshop creators see old dashboard links' do
+    teacher = create :teacher, terms_of_service_version: 1
+    teacher.permission = UserPermission::CREATE_PROFESSIONAL_DEVELOPMENT_WORKSHOP
+    sign_in teacher
+    get :home
+    assert_select 'h1', text: 'Old CSF Workshop Dashboard'
+  end
+
+  test 'teachers cannot see dashboard links' do
+    sign_in create(:teacher, terms_of_service_version: 1)
+    get :home
+    assert_select 'h1', false, text: 'Workshop Dashboard'
```

Comment:
The `false` 2nd argument makes it a negative assertion. Unfortunately [assert_select](http://api.rubyonrails.org/v4.1/classes/ActionDispatch/Assertions/SelectorAssertions.html#method-i-assert_select) does not follow the typical minitest assertion pattern and does not have a matching `refute_select` :(